### ISSUE TITLE
Volunteer link does nothing and gets a js error

### ISSUE THREAD

**Issue by tonysurma** [HTBox/allReady#884]
http://allready-d.azurewebsites.net/Event/5

![js error](https://cloud.githubusercontent.com/assets/5067595/16131165/6a8e1f04-33d2-11e6-9770-89a8fd8bca56.png)

quick note to list it and steve is looking at it

**Comment by stevejgordon** [comment]
So far I've been able to replicate this locally by setting up an event with a task that has a skill linked to it. If you then try to click volunteer and have no skills assigned to you, this error seems to occur.

I then added a matching skill to my user and tried again. This time the event details page blows up with a Json.Net self referencing error. This seems to be because the EventViewModel which is trying to serialise uses the domain/EF models which cause the self referencing. I'm playing with a cut down skill model for the userSkills which has helped at least get the page loading, but the event.js still seems to be doing some stuff with knockout that causes other JS errors.

So long story short, I'm out of my depth with the JS stuff. I'll keep looking at this is pretty critical to the functionality, but if anyone else has more JS experience and can take a look / assist please do. Just leave a comment here so I know I can give up!

**Comment by dpaquette** [comment]
I think I fixed this problem although I can't say for certain whether or not it is currently implemented they way it was intended. The problem was because the SkillViewModel class was missing an Id property. I have added that and now the "Matching" tag shows up as expected if your skills match the skills required for the task. The user also able to Volunteer without errors.

**Comment by tonysurma** [comment]
So I can volunteer now - good news.
I can't go to a task after I volunteered - bad news.


![js error 2](https://cloud.githubusercontent.com/assets/5067595/16135964/84ae51e6-33ec-11e6-8aa3-13fc04fa5dc0.png)

**Comment by dpaquette** [comment]
This can be closed, right?

**Comment by tonysurma** [comment]
thought I closed it already
